src/context/ContextCalendar.jsx

```diff
@@ -9,12 +16,45 @@ class ContextCalendar extends React.Component {
         this.props.onSelected(this.props.state, context);
     }
 
+    onKeyDown = (e, data, state, editor) => {
+        const keyCode = e.which;
+        if (keyCode === ENTER_KEY) {
+            e.preventDefault();
+            e.stopPropagation();
+            if (Lang.isUndefined(this.cal.state.selectedValue)) {
+                this.cal.state.selectedValue = this.cal.state.value.format("D MMM YYYY");
+            }
+            this.props.closePortal();
+            const date = this.cal.state.selectedValue;
+            const context = { key: 'set-date-id', context: `${date}`, object: date };
+            // If a plugin returns a state, it short circuits future plugins
+            return this.props.onSelected(this.props.state, context);
```

Comment:
[DP3?] Instead of replicating the functionality in handleDateSelect, is there any way we could call handleDateSelect directly? It looks like there is a subtle difference in that the original function didn't have a return value, but I'm wondering if anything would go awry if we added a return in that function. Could we explore?